### PR TITLE
MAINT: Hard tab and whitespace cleanup.

### PR DESCRIPTION
**Pull request by charris**
Mostly fixing hard tabs that have crept into the source files. The generated lapack_lite files also have a lot of hard tabs, but those are left unchanged.

### CODE REVIEW

File: numpy/core/src/multiarray/alloc.c [hunk 1]
```diff
@@ -263,7 +263,7 @@ PyDataMem_RENEW(void *ptr, size_t size)
 
     result = realloc(ptr, size);
     if (result != ptr) {
-	PyTraceMalloc_Untrack(NPY_TRACE_DOMAIN, (npy_uintp)ptr);
+    PyTraceMalloc_Untrack(NPY_TRACE_DOMAIN, (npy_uintp)ptr);
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/alloc.c]: This should be 8 spaces

**Review by eric-wieser — APPROVED** [review]
One nit, but I guess still a clear improvement anyway

**Comment by charris** [comment]
Fixed.

**Comment by charris** [comment]
@eric-wieser Thanks for the review.

**Pull request merged**